models/DataObject/ClassDefinition/Data/ReverseManyToManyObjectRelation.php

```diff
@@ -8,264 +8,25 @@
-     *
-     * @return $this
-     */
-    public function setOwnerFieldName($fieldName)
-    {
-        $this->ownerFieldName = $fieldName;
-
-        return $this;
-    }
-
-    /**
-     *
-     * Checks if an object is an allowed relation
-     *
-     * @param DataObject\Concrete $object
-     *
-     * @return bool
-     */
-    protected function allowObjectRelation($object)
-    {
-        //only relations of owner type are allowed
-        $ownerClass = DataObject\ClassDefinition::getByName($this->getOwnerClassName());
-        if ($ownerClass instanceof DataObject\ClassDefinition && $object instanceof DataObject\Concrete && $ownerClass->getId() == $object->getClassId()) {
-            $fd = $ownerClass->getFieldDefinition($this->getOwnerFieldName());
-            if ($fd instanceof DataObject\ClassDefinition\Data\ManyToManyObjectRelation) {
-                return $fd->allowObjectRelation($object);
-            }
-        }
-
-        return false;
-    }
-
-    /**
-     * Checks if data is valid for current data field
-     *
-     * @param mixed $data
-     * @param bool $omitMandatoryCheck
-     *
-     * @throws \Exception
-     */
-    public function checkValidity($data, $omitMandatoryCheck = false)
-    {
-        //TODO
-        if (!$omitMandatoryCheck and $this->getMandatory() and empty($data)) {
-            throw new Model\Element\ValidationException('Empty mandatory field [ '.$this->getName().' ]');
-        }
-
-        if (is_array($data)) {
-            foreach ($data as $o) {
-                $allowClass = $this->allowObjectRelation($o);
-                if (!$allowClass or !($o instanceof DataObject\Concrete)) {
-                    throw new Model\Element\ValidationException('Invalid non owner object relation to object ['.$o->getId().']', null, null);
-                }
-            }
-        }
-    }
-
-    /**
-     * converts object data to a simple string value or CSV Export
-     *
-     * @abstract
-     *
-     * @param DataObject\Concrete $object
-     * @param array $params
-     *
-     * @return string
-     */
-    public function getForCsvExport($object, $params = [])
-    {
-        return '';
-    }
-
-    /**
-     * fills object field data values from CSV Import String
-     *
-     * @param string $importValue
-     * @param null|DataObject\Concrete $object
-     * @param mixed $params
-     *
-     * @return null
-     */
-    public function getFromCsvImport($importValue, $object = null, $params = [])
-    {
-        return null;
-    }
-
-    /**
-     * This is a dummy and is mostly implemented by relation types
-     *
-     * @param mixed $data
-     * @param array $tags
-     *
-     * @return array
-     */
-    public function getCacheTags($data, $tags = [])
-    {
-        return $tags;
-    }
+@trigger_error(
+    'Data-type `Pimcore\Model\DataObject\ClassDefinition\Data\ReverseManyToManyObjectRelation` is deprecated since version 6.0.0 and will be removed in Pimcore 10. ' .
```

Comment:
```suggestion
    'Data-type `Pimcore\Model\DataObject\ClassDefinition\Data\ReverseManyToManyObjectRelation` is deprecated since version 6.9.0 and will be removed in Pimcore 10. ' .
```